app/questionnaire/questionnaire_store_updater.py

```diff
@@ -81,34 +72,44 @@ def mark_relationship_question_incomplete(self, list_name: str) -> None:
         )
         if target_relationship_collectors:
             for target in target_relationship_collectors:
-                self.remove_completed_location(Location(self._schema.get_section_for_block_id(target['id'])['id'], target['id']))
+                self.remove_completed_location(
+                    Location(
+                        self._schema.get_section_for_block_id(target['id'])['id'],
+                        target['id'],
+                    )
+                )
 
     def assess_relationship_question_completeness(self, list_name: str) -> None:
         target_relationship_collectors = self.get_relationship_collectors_by_list_name(
             list_name
         )
 
-        if target_relationship_collectors:
-            for target in target_relationship_collectors:
-                list_items = self._list_store.get(list_name).items
-                expected_pairs = [pair for pair in combinations(list_items, 2)]
-                relationship_answer_id = self._schema.get_relationship_answer_id_for_block(
-                    target['id']
-                )
-                relationship_answers = self._check_relationships_in_answer_store(
-                    relationship_answer_id
-                )
-
-                if relationship_answers:
-                    pairs = [
-                        (answer['list_item_id'], answer['to_list_item_id'])
-                        for answer in relationship_answers
-                    ]
-
-                    if expected_pairs == pairs:
-                        self.add_completed_location(
-                                Location(self._schema.get_section_for_block_id(target['id'])['id'], target['id'])
+        if not target_relationship_collectors:
+            return None
```

Comment:
Is this needed since the for loop exits on the first loop?

Reply:
because there may be no relationship collectors for a given list name